entries/data.xml

```diff
@@ -76,46 +76,42 @@ $( "span:last" ).text( $( "div" ).data( "test" ).last );
-alert( $( "body" ).data( "foo" ) );
-alert( $( "body" ).data() );
+var elem = document.createElement( "span" );
+$( elem ).data( "foo" ); // undefined
+$( elem ).data(); // {}
+
+$( elem ).data( "foo", 42 );
+$( elem ).data( "foo" ); // 42
+$( elem ).data(); // { foo: 42 }
       </code></pre>
-      <p>The above lines alert the data values that were set on the <code>body</code> element. If no data at all was set on that element, <code>undefined</code> is returned.</p>
+      <p>Calling <code>.data()</code> with no parameters returns a JavaScript object containing each stored value as a property. Using the object directly to get or set values is faster than making individual calls to <code>.data()</code>:</p>
       <pre><code>
-alert( $( "body" ).data( "foo" ) ); // undefined
-$( "body" ).data( "bar", "foobar" );
-alert( $( "body" ).data( "bar" ) ); // foobar
-      </code></pre>
-      <p><strong>jQuery 3</strong> changes the behavior of this method to align it to the <a href="https://www.w3.org/TR/html5/dom.html#dom-dataset">Dataset API specifications</a>. Specifically, jQuery 3 transforms every two-character sequence of "-" (U+002D) followed by a lowercase ASCII letter by the uppercase version of the letter as per definition of <a href="https://www.w3.org/TR/html5/dom.html#dom-dataset">the algorithm of the Dataset API</a>. Writing a statement like <code>$( "body" ).data( { "my-name": "aValue" } ).data();</code> will return <code>{ myName: "aValue" }</code>.</p>
+var mydata = $( "#mydiv" ).data();
+mydata.count = 43;
+mydata.status = "embiggened";
+$( "#mydiv" ).data( "count" ); // 43
+    </code></pre>
+      <p>Since <strong>jQuery 3</strong>, every two-character sequence of "-" (U+002D) followed by a lowercase ASCII letter in a key is replaced by the uppercase version of the letter, in alignment with the <a href="https://html.spec.whatwg.org/multipage/dom.html#dom-dataset">HTML dataset API</a>. A statement like <code>$( "body" ).data( { "my-name": "aValue" } ).data();</code> will return <code>{ myName: "aValue" }</code>.</p>
       <h4 id="data-html5">
         <a href="#data-html5">HTML5 data-* Attributes</a>
       </h4>
-      <p>As of jQuery 1.4.3 <a href="https://johnresig.com/blog/html-5-data-attributes/">HTML 5 data- attributes</a> will be automatically pulled in to jQuery's data object. The treatment of attributes with embedded dashes was changed in jQuery 1.6 to conform to the <a href="https://www.w3.org/TR/html5/dom.html#embedding-custom-non-visible-data-with-the-data-*-attributes">W3C HTML5 specification</a>.</p>
+      <p>Since jQuery 1.4.3, <a href="https://html.spec.whatwg.org/multipage/dom.html#embedding-custom-non-visible-data-with-the-data-*-attributes">data-* attributes</a> are used to initialize jQuery data. An element's data-* attributes are retrieved the first time the <code>data()</code> method is invoked upon it, and then are no longer accessed or mutated (all values are stored internally by jQuery).</p>
```

Comment:
What do you think about wrapping all `data-*` occurrences in code blocks?

Reply:
There is precedent for both `<code>`-wrapped and unwrapped attribute names throughout the API documentation, including both patterns on the same page in e.g. [`.attr()`](https://api.jquery.com/attr/). But I also like it better with wrapping. Done.